Dockerfile

```
@@ -2,10 +2,12 @@
 ARG TARGET_GOOS
 ARG TARGET_GOARCH
 FROM golang:1.17.1 as builder
+ARG TARGET_GOOS
+ARG TARGET_GOARCH
 ENV GO111MODULE=on \
     CGO_ENABLED=0 \
-    TARGET_GOOS=${TARGET_GOOS} \
-    TARGET_GOARCH=${TARGET_GOARCH}
+    TARGET_OS=${TARGET_GOOS} \
+    TARGET_ARCH=${TARGET_GOARCH}
```

```suggestion
    GOOS=${TARGET_GOOS} \
    GOARCH=${TARGET_GOARCH}
```
I think that instead of setting the `TARGET_OS` and `TARGET_ARCH` env vars in here, we should set the `GOOS` and `GOARCH` env vars as that is what is getting used primarily in the Makefile which in turn are getting used to set `TARGET_OS` and `TARGET_ARCH`.
https://github.com/cloudflare/cloudflared/blob/1733fe8c656c6104547684367f3a70899e790402/Makefile#L53-L57
https://github.com/cloudflare/cloudflared/blob/1733fe8c656c6104547684367f3a70899e790402/Makefile#L76-L80
So, directly setting `TARGET_OS` and `TARGET_ARCH` might have some side effects.